src/odemis/acq/fastem.py

```diff
@@ -705,20 +705,35 @@ def pre_calibrate(self, pre_calibrations):
         logging.debug(f"Moving to stage position x: {pos_hor}, y: {pos_vert}")
         self._stage_scan.moveAbsSync({'x': pos_hor, 'y': pos_vert})
 
-        self._pre_calibrations_future = align(self._scanner, self._multibeam,
-                                              self._descanner, self._detector,
-                                              self._stage, self._ccd,
-                                              self._beamshift, None,  # no need for the detector rotator
-                                              self._se_detector, self._ebeam_focus,
-                                              calibrations=pre_calibrations)
-
         try:
-            self._pre_calibrations_future.result()  # wait for the calibrations to be finished
-        except CancelledError:
-            logging.debug("Cancelled acquisition pre-calibrations.")
-            raise
+            for i in range(3):  # try running the pre-calibrations 3 times
+                logging.debug(f"Will run pre-calibrations at field index {self.field_idx}")
+                try:
+                    self._pre_calibrations_future = align(self._scanner, self._multibeam,
+                                                          self._descanner, self._detector,
+                                                          self._stage, self._ccd,
+                                                          self._beamshift, None,  # no need for the detector rotator
+                                                          self._se_detector, self._ebeam_focus,
+                                                          calibrations=pre_calibrations)
+                    self._pre_calibrations_future.result()  # wait for the calibrations to be finished
+                    break  # if it successfully ran, do not try again
+                except CancelledError:
+                    logging.debug("Cancelled acquisition pre-calibrations.")
+                    raise
+                except Exception as err:
+                    logging.warning(f"Pre-calibration failed for ROA {self._roa.name.value} with error {err}, "
+                                    f"will try again.")
```

Comment:
This warning will also be shown when i == 2. That's a bit confusing, because it won't try again. Maybe place this warning as a `else` of the `if == 2`, to only show it in the first 2 trials.